packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,384 @@
+  public mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    renderingEngine: any;
+    annotations: Map<string, Annotation>;
+  } | null = {} as any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        sourceViewportIds: [],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  _init = (): void => {
+    const renderingEngines = getRenderingEngines();
+    const renderingEngine = renderingEngines[0];
+
+    // Todo: handle this case where it is too soon to get the rendering engine
+    if (!renderingEngine) {
+      return;
+    }
+
+    let viewports = renderingEngine.getViewports();
+    viewports = filterViewportsWithToolEnabled(viewports, this.getToolName());
+
+    // const sourceViewport = renderingEngine.getViewport(
+    //   this.configuration.sourceViewportId
+    // ) as Types.IVolumeViewport;
+
+    this.configuration.sourceViewportIds.forEach((viewportId) => {
+      const sourceViewport = renderingEngine.getViewport(viewportId);
+      if (sourceViewport || sourceViewport.getImageData()) {
+        const { element } = sourceViewport;
+        const { viewUp, viewPlaneNormal } = sourceViewport.getCamera();
+
+        const FrameOfReferenceUID = sourceViewport.getFrameOfReferenceUID();
+        if (!this.editData?.annotations) {
+          this.editData.annotations = new Map();
+        }
+
+        let annotation = this.editData.annotations[viewportId];
+
+        if (!annotation) {
+          const newAnnotation: OverlayGridAnnotation = {
+            highlighted: true,
+            invalidated: true,
+            metadata: {
+              toolName: this.getToolName(),
+              viewPlaneNormal: <Types.Point3>[...viewPlaneNormal],
+              viewUp: <Types.Point3>[...viewUp],
+              FrameOfReferenceUID,
+              referencedImageId: null,
+            },
+            data: {
+              sourceViewport,
+            },
+          };
+
+          addAnnotation(newAnnotation, element);
+          annotation = newAnnotation;
+        }
+
+        this.editData.annotations.set(viewportId, annotation);
+        this.editData.renderingEngine = renderingEngine;
+      }
+    });
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      viewports.map((viewport) => viewport.id)
+    );
+  };
+
+  onSetToolEnabled = (): void => {
+    this._init();
+  };
+
+  onSetToolActive = (): void => {
+    this._init();
+  };
+
+  onCameraModified = (evt: Types.EventTypes.CameraModifiedEvent): void => {
+    // If the camera is modified, we need to update the reference lines
+    // we really don't care which viewport triggered the
+    // camera modification, since we want to update all of them
+    // with respect to the targetViewport
+    this._init();
+  };
```

Comment:
the comment is wrong, and in fact we care about not updating the viewports whose camera is not updated. If we only zoom in one of the target viewports, why the other ones (from target) should update their reference lines